k8s.io/deployment-short-urls.yaml

```diff
@@ -0,0 +1,58 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: short-urls
+spec:
+  selector:
+    matchLabels:
+      app: short-urls
+  template:
+    metadata:
+      labels:
+        app: short-urls
+    spec:
+      containers:
+      - name: shlink
+        image: shlinkio/shlink:1.16.2
+        command: ["/bin/sh", "-c"]
+        args:
+        - |
+          # Start shlink server in the background.
+          /bin/sh /etc/shlink/docker-entrypoint.sh &
+          PID=$!
+          # Wait for service to become available.
+          while ! shlink short-url:list; do
+            sleep 10s
+          done
+          # Load URLs from the config map.
+          while read slug url; do
+            shlink short-url:generate "--customSlug=${slug}" "${url}"
+          done < /etc/config/short-urls
+          # Create healthcheck URL for our readiness probe.
+          shlink short-url:generate -c shlink-healthcheck https://k8s.io
```

Comment:
I don't grok what this is doing?  is it making a redirect to k8s.io as a way to prove readiness?  Very soon we won't follow the URL at all but just accept 301/302 as succes, which still seems OK

Reply:
This entry is added to the database after all the "real" entries are added. This is because the shlink service is up and running, but I didn't want the readiness probe to be OK until we finish loading config.

Reply:
In my mind this was going to help with deployment rollouts when we have more than 1 replica, so that "ready" instances have all the short URLs loaded. However, since the container is restarting inside the pod, I'm not sure if the deployment controller is capable of doing the right thing.

Reply:
If, for example, shlink deadlocks the readiness probe will fail, and it will be removed from LBs.  You can probably use this as both a readiness and a liveness probe with different timings, so it doesn't become ready until all URLs are loaded and if it stops serving kubelet will kill it.